SU-HSDO/suhumsci
# STN-568: Views Dark Color Inversion

# READY FOR REVIEW

## Summary
Use an existing pairing to reverse color on specific elements in Colorful and Traditional themes.
This PR should apply dark mode variants for views, using both the theme setting for dark inversion and the `hb-dark-pattern` utility class directly applied to the view.

JIRA Card: https://sparkbox.atlassian.net/browse/STN-568
Figma Link: [(Upcoming Events & Seminars Section)](https://www.figma.com/file/toNZnFLadtpNpG5dPzNGWz/%E2%9A%99%EF%B8%8F-Colorful-Theme---Working?node-id=248%3A0)

## Need Review By (Date)
12/16/20

## Urgency
low

**For this PR you will need to test both the Theme Dark Inversion setting (which can be found at Appearance > [Theme] > Settings) and the `hb-dark-pattern` utility class for any views you are testing on.**

## Steps to Test
1. `npm run test`
2. To test the dark mode color inversion toggle for views.
- Switch on the dark inversion setting in the theme appearance settings.
- From here you will need to test a few different view types, such as: events, news, people, publications, courses and research, etc.
- Some pages you can use to view these views listed below:
http://sparkbox-sandbox.suhumsci.loc/qa-views (`hb-pill`s seems to be broken in light and dark mode, not from work done on this card.)
_Courses:_
http://sparkbox-sandbox.suhumsci.loc/courses
http://sparkbox-sandbox.suhumsci.loc/courses-0
_Events:_
http://sparkbox-sandbox.suhumsci.loc/events
http://sparkbox-sandbox.suhumsci.loc/events/past-events
_People:_
http://sparkbox-sandbox.suhumsci.loc/people/people-flexible-image-classes
http://sparkbox-sandbox.suhumsci.loc/people/people-grid
_News:_
http://sparkbox-sandbox.suhumsci.loc/example-content/news
http://sparkbox-sandbox.suhumsci.loc/news
_Publications:_
http://sparkbox-sandbox.suhumsci.loc/publications/book-shelf
http://sparkbox-sandbox.suhumsci.loc/publications/publication-table
_Seminars:_
http://sparkbox-sandbox.suhumsci.loc/publications/book-shelf

- You can also create your own page and add Views blocks in manually through Paragraphs, if you'd like to test that way as well.
3. Once you are done testing different views with the toggle setting on the first theme type, you'll want to swap it over to Colorful or Traditional now and test those pages again depending on which theme you tested first.
4. To test `hb-dark-pattern` utility class in views individually. (make sure the toggle for the theme is turned off)
- To add the utility class you'll need to find the first view you'd like to test and click the "Edit View" contextual link for it.
- From there add the `hb-dark-pattern` class to the view by going to the "CSS class:" setting under the Views "Other:" section.
- Once you've added the class, click Apply on the setting window, then Save the view.
- The view should now look the same as what it looked like while you were testing the Dark Mode theme setting in the appearance settings.
- Repeat these steps for any additional views you'd like to test using the list above or the flexible page you may have created.
- Check the views in both Colorful and Traditional themes and color pairings.

**Note: The site search view was not tested or styled as it is specifically called out in the card as not included in this work.**

## PR Checklist
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
- [Sparkbox PR Checklist](../docs/SparkboxPRChecklist.md)
